app/controllers/buddies_controller.rb

```diff
@@ -16,7 +16,7 @@ def create
       flash[:notice] = "You've successfully submitted a buddy!"
       redirect_to @buddy
     else
-      flash[:alert] = "Fail"
+      flash[:alert] = @buddy.errors.full_messages
```

Comment:
This returns an array of strings. You might want to format it as a string before storing it in flash.